File: java/src/test/java/com/libutil/util/Base64Test.java
package com.libutil.util;

import com.libutil.Log;
import com.libutil.Util;

public class Base64Test {

  public static void main(String args[]) {
    encodeBase64Test();
    decodeBase64Test();
  }

  private static void encodeBase64Test() {
    String str = "abc";
    Log.d("encodeBase64(\"" + str + "\")");
    String encoded = Util.encodeBase64(str);
    Log.d(encoded);

    str = "";
    Log.d("encodeBase64(\"" + str + "\")");
    encoded = Util.encodeBase64("");
    Log.d(encoded);

    str = null;
    Log.d("encodeBase64(null)");
    encoded = Util.encodeBase64("");
    Log.d(encoded);

    byte[] b = { 97, 98, 99 };
    Log.d("encodeBase64([97,98,99]");
    encoded = Util.encodeBase64(b);
    Log.d(encoded);
  }

  private static void decodeBase64Test() {
    String str = "YWJj";
    Log.d("decodeBase64(\"" + str + "\")");
    String decoded = Util.decodeBase64(str);
    Log.d(decoded);

    str = "";
    Log.d("decodeBase64(\"" + str + "\")");
    decoded = Util.decodeBase64(str);
    Log.d(decoded);

    str = null;
    Log.d("decodeBase64(null)");
    decoded = Util.decodeBase64(str);
    Log.d(decoded);

    str = "YWJj";
    Log.d("decodeBase64B(\"" + str + "\")");
    byte[] d = Util.decodeBase64B(str);
    for (int i = 0; i < d.length; i++) {
      Log.d(d[i]);
    }
  }

}

File: python/test/misc/base64_test.py
#!python

import os
import sys

sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
import util

def test_decode_base64():
    s = '\n'
    s += util.decode_base64('YWJj')
    s += '\n'
    s += util.decode_base64('YWJj', 'utf-8')
    s += '\n'
    s += util.decode_base64('YWJj', 'shift-jis')
    s += '\n'
    s += util.decode_base64('YWJj44GC44GE44GG', 'utf-8')
    s += '\n'
    s += util.decode_base64('YWJjgqCCooKk', 'shift-jis')
    return s

def test_encode_base64():
    s = '\n'
    s += util.encode_base64('abc')
    s += '\n'
    s += util.encode_base64('abc', 'utf-8')
    s += '\n'
    s += util.encode_base64('abc', 'shift-jis')
    s += '\n'
    s += util.encode_base64('abcあいう', 'utf-8')
    s += '\n'
    s += util.encode_base64('abcあいう', 'shift-jis')
    return s

def test():
    ret = ''
    ret += 'test_decode_base64() = ' + test_decode_base64() + '\n'
    ret += '\n'
    ret += 'test_encode_base64() = ' + test_encode_base64() + '\n'
    return ret

def main():
    try:
        ret = test()
    except Exception as e:
        ret = str(e)
    util.send_response('text', ret)

main()
